File: apps/modernization-api/src/main/java/gov/cdc/nbs/patient/profile/summary/PatientSummary.java
package gov.cdc.nbs.patient.profile.summary;

import java.time.LocalDate;
import java.util.Collection;

public record PatientSummary(
    LocalDate asOf,
    long patient,
    Name legalName,
    LocalDate birthday,
    Integer age,
    String gender,
    String ethnicity,
    Collection<Phone> phone,
    Collection<Email> email) {

  record Name(
      String prefix,
      String first,
      String middle,
      String last,
      String suffix) {
  }


  record Phone(
      String use,
      String number) {
  }


  record Email(
      String use,
      String address) {
  }


  record PatientSummaryIdentification(
      String type,
      String value) {
  }
}

File: apps/modernization-ui/src/patient/profile/summary/PatientSummary.ts
type Name = {
    prefix: string | null;
    first: string | null;
    middle: string | null;
    last: string | null;
    suffix: string | null;
};

type Phone = {
    use: string;
    number: string;
};

type Email = {
    use: string;
    address: string;
};

type Address = {
    street: string | null;
    city: string | null;
    state: string | null;
    zipcode: string | null;
    country: string | null;
};

export type PatientSummary = {
    patient: number;
    legalName: Name;
    birthday: string;
    age: number;
    gender: string | null;
    ethnicity: string | null;
    race: string | null;
    phone: Phone[];
    email: Email[];
    address: Address | null;
};
